packages/trader/src/Stores/Modules/MT5/mt5-store.js

```diff
@@ -160,31 +164,37 @@ export default class MT5Store extends BaseStore {
     @action.bound
     beginRealSignupForMt5() {
         sessionStorage.setItem('post_real_account_signup', JSON.stringify(this.account_type));
-        this.root_store.ui.is_real_acc_signup_on = true;
+        this.root_store.ui.openRealAccountSignup();
     }
 
     realMt5Signup() {
-        // Check if the user has real account
-        if (!this.root_store.client.has_active_real_account
-            || !this.root_store.client.currency
-        ) {
-            this.beginRealSignupForMt5();
-        } else {
-            switch (this.account_type.type) {
-                case 'standard':
-                    this.enableMt5PasswordModal();
-                    break;
-                case 'advanced':
-                    // eslint-disable-next-line no-console
-                    console.log('Open Real advanced account for user');
-                    break;
-                case 'synthetic_indices':
-                    // eslint-disable-next-line no-console
-                    this.enableMt5PasswordModal();
-                    break;
-                default:
-                    throw new Error('Cannot determine mt5 account signup.');
-            }
+        switch (this.account_type.type) {
+            case 'standard':
+                this.enableMt5PasswordModal();
+                break;
+            case 'advanced':
+                this.root_store.client.fetchResidenceList();
+                this.root_store.client.fetchStatesList();
+                this.root_store.client.fetchAccountSettings();
+                this.enableMt5AdvancedModal();
+                break;
+            case 'synthetic_indices':
+                // eslint-disable-next-line no-console
+                this.enableMt5PasswordModal();
+                break;
+            default:
+                throw new Error('Cannot determine mt5 account signup.');
+        }
+    }
+
+    @action.bound
+    enableMt5AdvancedModal() {
+        if (this.account_type.category === 'real' && this.account_type.type === 'advanced') {
+            this.is_mt5_advanced_modal_open = true;
+            // this.root_store.ui.setRealAccountSignupParams({
+            //     active_modal_index: 6, // Real MT5 Advanced Modal wizard
+            // });
+            // this.root_store.ui.is_real_acc_signup_on = true;
```

Comment:
Oh yeah, since extracting out of the MT5 advanced flow from the real account signup, I forgot to remove this section of the code. it is not necessary anymore. the mt5 signup flow has its own wizard component now.